ruma-federation-api/src/query/custom/v1.rs

```diff
@@ -0,0 +1,45 @@
+//! [GET /_matrix/federation/v1/query/{queryType}](https://matrix.org/docs/spec/server_server/r0.1.4#get-matrix-federation-v1-query-querytype)
+
+use ruma_api::ruma_api;
+use serde_json::Value as JsonValue;
+
+ruma_api! {
+    metadata: {
+        description: "Performs a single query request on the receiving homeserver. The query string arguments are dependent on which type of query is being made.",
+        method: GET,
+        name: "custom",
+        path: "/_matrix/federation/v1/query/:kind",
+        rate_limited: false,
+        authentication: AccessToken,
+    }
+
+    request: {
+        /// The type of query to make.
+        #[ruma_api(path)]
+        pub kind: &'a str,
```

Comment:
Why not keep the spec naming, i.e. `query_type`?

Reply:
I think I just saw type in the name and changed it to kind on autopilot. `query_type` is definitely clearer